libs/designer/src/lib/ui/panel/connectionsPanel/createConnection/createConnectionWrapper.tsx

```diff
@@ -127,6 +128,12 @@ export const CreateConnectionWrapper = () => {
     dispatch(openPanel({ nodeId, panelMode }));
   }, [dispatch, referencePanelMode, nodeIds]);
 
+  const queryClient = useQueryClient();
+  const useUpdateNewConnection = useMutation(async (newConnection: Connection) => {
+    return queryClient.setQueryData<Connection[]>(['connections', connector?.id?.toLowerCase()], (oldConnections) => {
```

Comment:
since the only mutation is updating query data, this may be able to be a useCallback and skip the mutation entirely. Mutation infers that there is a server state change.